### PR TITLE
Add relocation information

### LINKED ISSUE #7968: Maven distribution relocation for Track-2 GAs
For each GA'd Track-2 project, setup a relocation notification for the move from (Track-1) `com.microsoft.azure` to `com.azure` (Track-2 / next major release)

example:
https://mvnrepository.com/artifact/org.codehaus.jackson/jackson-mrbean

cc: @JimSuplizio @JonathanGiles

### PR DESCRIPTION
**Pull request by conniey**
* Adds relocation information in track 1 client libraries.
  * Event Hubs
  * Key Vault
  * Storage
* See: [Guide to relocation](https://maven.apache.org/guides/mini/guide-relocation.html)
  * [POM relocation model](https://maven.apache.org/ref/3.6.3/maven-model/maven.html#class_relocation)

Fixes #7968

### CODE REVIEW

File: sdk/eventhubs/microsoft-azure-eventhubs-eph/pom.xml [hunk 1]
```diff
@@ -25,6 +25,20 @@
       <id>azure-java-build-docs</id>
       <url>${site.url}/site/${project.artifactId}</url>
     </site>
+    <relocation>
```
Comment (srnagar) [inline sdk/eventhubs/microsoft-azure-eventhubs-eph/pom.xml]: Doesn't this require a version update too? In this case `3.1.1`?

File: sdk/keyvault/microsoft-azure-keyvault-cryptography/pom.xml [hunk 2]
```diff
@@ -21,10 +21,25 @@
...
-            <id>azure-java-build-docs</id>
-            <url>${site.url}/site/${project.artifactId}</url>
-        </site>
+      <site>
+          <id>azure-java-build-docs</id>
+          <url>${site.url}/site/${project.artifactId}</url>
+      </site>
+      <relocation>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-security-keyvault-keys</artifactId>
```
Comment (srnagar) [inline sdk/keyvault/microsoft-azure-keyvault-cryptography/pom.xml]: Can relocation section have multiple artifacts? We can include keys, secrets and certificates if that's possible.

File: sdk/eventhubs/microsoft-azure-eventhubs-eph/pom.xml (hunk 1 above)
Comment (conniey) [reply sdk/eventhubs/microsoft-azure-eventhubs-eph/pom.xml]: 3.1.0 hasn't been released yet. The latest is 3.0.2, i believe the current versions are correct.

File: sdk/keyvault/microsoft-azure-keyvault-cryptography/pom.xml (hunk 2 above)
Comment (conniey) [reply sdk/keyvault/microsoft-azure-keyvault-cryptography/pom.xml]: I checked the pom definition, it doesn't look defined.

**Pull request merged**